src/connection.ts

```diff
@@ -1917,7 +1924,10 @@ class Connection extends EventEmitter {
   initialiseConnection() {
     const signal = this.createConnectTimer();
 
-    if (this.config.options.port) {
+    if (this.config.options.connector) {
+      // port and multiSubnetFailover are not used when using a custom connector
+      return this.connectOnPort(0, false, signal, this.config.options.connector);
+    } else if (this.config.options.port) {
       return this.connectOnPort(this.config.options.port, this.config.options.multiSubnetFailover, signal, this.config.options.connector);
     } else {
       return instanceLookup({
```

Comment:
I think this is the place you need to change. We know here that `this.config.server` is not `undefined` in this `else` block (even if the type is declared as `string | undefined`), so we can tell TypeScript that this is the case by changing `this.config.server` to `this.config.server!`.